spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/observation/ObservationAutoConfiguration.java

```diff
@@ -90,8 +88,16 @@ OnlyMetricsObservationHandlerGrouping onlyMetricsObservationHandlerGrouping() {
 	}
 
 	@Configuration(proxyBeanMethods = false)
-	@ConditionalOnClass(TracingObservationHandler.class)
-	static class TracingConfiguration {
+	@ConditionalOnBean(Tracer.class)
```

Comment:
This will match when `micrometer-core` isn't on the classpath. That will result in a failure as `TracingAwareMeterObservationHandler` implements `MeterObservationHandler` from `micrometer-core`. `@ConditionalOnClass(MeterRegistry.class)` should perhaps be added as well.